core/tracker/tracker.go

```diff
@@ -134,11 +149,55 @@ func analyseDutyFailed(duty core.Duty, es []event) (bool, component, string) {
 // TODO(xenowits): Implement logic for reporting duties.
 func failedDutyReporter(core.Duty, bool, string, string) {}
 
-// analyseParticipation returns the share indexes of peers that participated in this duty.
-// TODO(dhruv): implement logic to analyse participation.
-//nolint:deadcode
-func analyseParticipation(core.Duty, []event) []int {
-	return nil
+// analyseParticipation returns a set of share indexes of participated peers corresponding to each DV public key.
+func analyseParticipation(events []event) map[core.PubKey]map[shareIdx]bool {
+	// Set of shareIdx of participated peers per DV.
+	resp := make(map[core.PubKey]map[shareIdx]bool)
+
+	eventsPerDV := make(map[core.PubKey][]event)
+	for _, e := range events {
+		eventsPerDV[e.pubkey] = append(eventsPerDV[e.pubkey], e)
+	}
+
+	for pubKey, dvEvents := range eventsPerDV {
```

Comment:
I'm not sure about including pubkey in the whole analysis. Since we group pubkeys by duty, either all the pubkeys will be there that should be there for the peer or none will be there for the peer. It doesn't add any value. I would say we should exclude pubkey from this. Just do duty and peer, not pubkey.